src/libraries/Microsoft.PowerFx.Core/Types/DType.cs

```diff
@@ -827,11 +808,17 @@ public static DType CreateMetadataType(IDataColumnMetadata metadata)
             return new DType(DKind.Metadata, metadata, Unknown.TypeTree);
         }
 
+        /// <summary>
+        /// Attachment types can be either tables or records, and are represented using a LazyTable/Record type.
+        /// </summary>
         public static DType CreateAttachmentType(DType attachmentType)
         {
             Contracts.AssertValid(attachmentType);
-
-            return new DType(DKind.Attachment, attachmentType);
+            Contracts.Assert(attachmentType.IsAggregate);
+            
+            var attachmentRecord = new BuiltInLazyTypes.AttachmentType(attachmentType.ToRecord());
```

Comment:
>new BuiltInLazyTypes.AttachmentType(attachmentType.ToRecord()); [](http://example.com/codeflow?start=35&length=63)

Would following work?

var type = new BuiltInLazyTypes.AttachmentType( attachmentType.IsTable ? attachmentType : attachmentType.ToRecord());

return type._type;

Reply:
No, the backing type is always a RecordType, and we call ToTable on it if a table was actually what was expected.